File: src/modules/BitmovinCastManagerExpoModule.ts
import { NativeModule, requireNativeModule } from 'expo-modules-core';

export type BitmovinCastManagerExpoModuleEvents = Record<string, any>;

declare class BitmovinCastManagerExpoModule extends NativeModule<BitmovinCastManagerExpoModuleEvents> {
  isInitialized(): Promise<boolean>;
  initializeCastManager(options?: Record<string, any>): Promise<void>;
  sendMessage(message: string, messageNamespace?: string): Promise<void>;
  updateContext?(): Promise<void>; // Android only
}

export default requireNativeModule<BitmovinCastManagerExpoModule>(
  'BitmovinCastManagerModule'
);

File: android/src/main/java/com/bitmovin/player/reactnative/BitmovinCastManagerExpoModule.kt
package com.bitmovin.player.reactnative

import com.bitmovin.player.casting.BitmovinCastManager
import com.bitmovin.player.reactnative.converter.toCastOptions
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.types.Enumerable

class BitmovinCastManagerExpoModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("BitmovinCastManagerModule")
        
        AsyncFunction("isInitialized") {
            BitmovinCastManager.isInitialized()
        }
        
        AsyncFunction("initializeCastManager") { options: Map<String, Any>? ->
            val applicationId = options?.get("applicationId") as? String
            val messageNamespace = options?.get("messageNamespace") as? String

            BitmovinCastManager.initialize(
                applicationId,
                messageNamespace,
            )
        }
        
        AsyncFunction("sendMessage") { message: String, messageNamespace: String? ->
            BitmovinCastManager.getInstance().sendMessage(message, messageNamespace)
        }
        
        AsyncFunction("updateContext") {
            appContext.currentActivity?.let { activity ->
                BitmovinCastManager.getInstance().updateContext(activity)
            }
        }
    }
}